assets/mkEnv/env-from-lockfile.nix

```diff
@@ -0,0 +1,26 @@
+# Build an environment from a collection of packages
+{
+  lockfilePath ?
+    throw
+    "flox: You must provide the path to a lockfile.",
+  system ? builtins.currentSystem or "unknown",
+  ...
+}: let
+  lockfileContents = builtins.fromJSON (builtins.readFile lockfilePath);
+  nixpkgsFlake = builtins.getFlake lockfileContents.registry.inputs.nixpkgs.url;
+  pkgs = nixpkgsFlake.legacyPackages.${system};
+  # Convert manifest elements to derivations.
+  tryGetDrv = system: package: let
+    flake = builtins.getFlake package.${system}.url;
+    drv = builtins.foldl' (attrs: pathComponent: builtins.getAttr pathComponent attrs) flake package.${system}.path;
+  in
+    if builtins.isNull package.${system}
+    then null
+    else drv;
+  entries =
+    builtins.filter
+    (p: !builtins.isNull p)
+    (builtins.map (tryGetDrv system)
+      (builtins.attrValues lockfileContents.packages));
+in
+  pkgs.linkFarmFromDrvs "flox-env" entries
```

Comment:
note nonblocking: we currently have no record of any versioning for this file, which means we can't reproduce the environment. What do we do if someone builds an environment with a different version of env-from-lockfile? Generations are currently 1:1 with a specific store path, but with this change, building a specific generation of an environment with a different version of flox could result in a different store path

Reply:
That's a really good point, but this code is expected to live for <8 weeks so I'm not sure it's a problem we really need to worry about right now.